open_spiel/bots/gin_rummy/simple_bot.cc

```diff
@@ -0,0 +1,217 @@
+// Copyright 2019 DeepMind Technologies Ltd. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+#include <vector>
+
+#include "open_spiel/spiel.h"
+#include "open_spiel/spiel_utils.h"
+#include "open_spiel/spiel_bots.h"
+
+#include "open_spiel/bots/gin_rummy/simple_bot.h"
+#include "open_spiel/games/gin_rummy.h"
+#include "open_spiel/games/gin_rummy/gin_rummy_utils.h"
+
+namespace open_spiel {
+namespace gin_rummy {
+
+void SimpleBot::Restart() {
+  knocked_ = false;
+  next_actions_ = {};
+}
+
+Action SimpleBot::Step(const State& state) {
+  std::vector<float> observation;
```

Comment:
Why not use Observer? Then you don't need to update offsets (as much at least)

Reply:
Ah, I built the first version of this bot a while ago and just didn’t switch it over to use Observer. How does this help us update offset less often though? Don’t we still get a tensor that we have to iterate over? And it's only through the python interface that we can access pieces by name right?

Reply:
Yeah good point, I forgot it's not in C++ in master yet. Nevermind then!